data/store/mongo/mongo_data_set.go

```diff
@@ -296,6 +269,30 @@ func (d *DataSetRepository) ListUserDataSets(ctx context.Context, userID string,
 	return dataSets, nil
 }
 
+func (d *DataSetRepository) CreateUserDataSet(ctx context.Context, userID string, create *data.DataSetCreate) (*data.DataSet, error) {
+	dataSet := data.NewDataSet()
+	dataSet.Client = create.Client
```

Comment:
Nit: A common pattern for platform services is to have a constructor which sets all fields of a struct given a "create" struct. Is there a reason why this logic is in the store and not in a separate constructor?

Reply:
I looked at the platform code and it is a bit hit-or-miss for which way it is done. For example, the task store does it this way.